UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/Admin/RadioNavigationalWarningEditAdminTest .cs

```diff
@@ -0,0 +1,150 @@
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.WarningType = 3;
+
+            Assert.ThrowsAsync(Is.TypeOf<InvalidDataException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidReference_ThenReturnArgumentNullException()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.Reference = string.Empty;
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidSummary_ThenReturnArgumentNullException()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.Summary = string.Empty;
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+        [Test]
+        public void WhenEditRadioNavigationWarningsWithInValidContent_ThenReturnArgumentNullException()
+        {
+            _controller.TempData = _tempData;
+            _fakeEditRadioNavigationalWarningsAdmin = GetFakeEditRadioNavigationalWarningsAdmin();
+            _fakeEditRadioNavigationalWarningsAdmin.Content = string.Empty;
+
+            Assert.ThrowsAsync(Is.TypeOf<ArgumentNullException>(),
+                                async delegate { await _controller.Edit(_fakeEditRadioNavigationalWarningsAdmin.Id, _fakeEditRadioNavigationalWarningsAdmin); });
+        }
+
+        public static EditRadioNavigationalWarningsAdmin GetFakeEditRadioNavigationalWarningsAdmin()
+        {
+            return new EditRadioNavigationalWarningsAdmin()
+            {
+                Id = 9,
+                WarningType = 1,
+                Reference = "EditReferenceTest",
+                DateTimeGroup = new DateTime(2022, 12, 30),
+                Summary = "EditSummaryTest",
+                Content = "EditContentTest",
+                ExpiryDate = new DateTime(2099, 1, 1),
+                IsDeleted = "true"
+            };
+        }
+
+        public static RadioNavigationalWarning GetFakeCreateRadioNavigationalWarning()
```

Comment:
Local static methods should be declared as private.
```suggestion
        private static RadioNavigationalWarning GetFakeCreateRadioNavigationalWarning()
```

Reply:
changes done